packages/blocks/src/api/registration.js

```diff
@@ -303,11 +304,50 @@ export function registerBlockType( name, settings ) {
 		return;
 	}
 
+	validateVariations( settings );
+
 	dispatch( blocksStore ).addBlockTypes( settings );
 
 	return settings;
 }
 
+function validateVariations( settings ) {
```

Comment:
I'm not really sure where to put this. 😄 I'm also wondering why don't we `dispatch( blocksStore ).addBlockVariation( settings.variations )` instead of handling variations here:

https://github.com/WordPress/gutenberg/blob/6ad2a433769a4514fc52083e97aa47a0bc9edf07/packages/blocks/src/store/reducer.js#L139

For this reason we need to do the variations validation in two functions.

Reply:
Can we extract it to another PR? It isn't a blocker really and it requires some more thinking because you can register block variations also through the store directly. I don't remember exactly but most likely they are stored in two places in the store. It's similar to how block styles can be registered, so plugin authors could register styles or variations before the block gets registered so plugins wouldn't have to declare dependencies on each other.

Reply:
That makes sense. I didn't know it meant to be that way - registering variation even without an existing block type. This comes with its own challenges 😄 

I revert the commit.

Reply:
I double-checked, it's stored in one reducer but you can add variations with `ADD_BLOCK_TYPES` or `ADD_BLOCK_VARIATIONS` 😄